certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -178,30 +188,41 @@ func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshift
 	defer cancel()
 
 	installedCSVDone := make(chan string, 1)
+	contextTimeOut := make(chan error, 1)
 
 	// query API server for the installed CSV field of the created subscription
-	go func() {
+	go func(ctx context.Context) error {
 		defer close(installedCSVDone)
+		defer close(contextTimeOut)
+
 		for {
-			time.Sleep(2 * time.Second)
 			log.Debug("Waiting for Subscription.status.installedCSV to become ready...")
 			subs, _ := openshiftengine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
 			installedCSV := subs.Status.InstalledCSV
 			// if the installedCSV field is present, stop the querying
 			if len(installedCSV) > 0 {
 				log.Debug(fmt.Sprintf("Subscription.status.installedCSV is %s", installedCSV))
 				installedCSVDone <- installedCSV
+				return nil
 			}
 			log.Debug("Subscription.status.installedCSV is not set yet, retrying...")
+
+			select {
+			case <-ctx.Done():
+				log.Error("failed to fetch Subscription.status.installedCSV: ", ctx.Err())
+				contextTimeOut <- ctx.Err()
+				return ctx.Err()
```

Comment:
Where is this return being used?

Reply:
Per our discussion, this is being used to force the goroutine to stop the execution.